test/cmdlineTests/exp_base_literal/output

```diff
@@ -10,7 +10,8 @@ IR:
 /// @use-src 0:"exp_base_literal/input.sol", 1:"#utility.yul"
 object "C_81" {
     code {
-        /// @src 0:82:370
+        /// @src 0:82:370  contract C {...
```

Comment:
Something is adding additional newlines here

Reply:
Fixed. The helper in `Common.h` was not expected to print them while `AsmParser` is. To avoid adding an extra argument in `formatSourceLocationComment()` just to control newlines, some of the overloads now skip this newline while some do not. Which is not great but at least preserves the current behavior.